### PR TITLE
Add ./validate.php -m rrdcheck

### PR DESCRIPTION
**Pull request by khobbits**
This adds a little tool to list unreadable rrd files in your rrd folder.
```
Scanning 9656 rrd files in /opt/librenms/rrd...
[FAIL]    Error parsing "/opt/librenms/rrd/nexus-c/port-5.rrd" RRD ERROR: '/opt/librenms/rrd/nexus-c/port-5.rrd.old' is not an RRD file
[FAIL]    Error parsing "/opt/librenms/rrd/nexus-c/ucd_diskio-dm-1.rrd" RRD ERROR: mmaping file '/opt/librenms/rrd/nexus-c/ucd_diskio-dm-1.rrd': Invalid argument
[FAIL]    Error parsing "/opt/librenms/rrd/nexus-c/ucd_diskio-dm-0.rrd" RRD ERROR: mmaping file '/opt/librenms/rrd/nexus-c/ucd_diskio-dm-0.rrd': Invalid argument
Status: 9656/9656 - Complete
```

### CODE REVIEW

**Comment by f0o** [comment]
Looks good from a brief view. Will test it on the CI after lunch

**Comment by khobbits** [comment]
To test the tool with output, just create some empty rrd files using touch.

File: includes/functions.php [hunk 1]
```diff
@@ -1366,3 +1366,69 @@ function dnslookup($device,$type=false,$return=false) {
...
+        ),
+        $pipes
+    );
+
+    if (!is_resource($process)) {
+        throw new \RuntimeException('Could not create a valid process');
+    }
+
+    $status = proc_get_status($process);
+    while($status['running']) {
```
Comment (f0o) [inline includes/functions.php]: This loop is very cpu heavy, please add a sleep in it. A couple of ms is fine

Comment (khobbits) [inline includes/functions.php]: Did a couple of benchmarks here:
Anything below 2ms increases the cpu usage time without speeding up the process. 
Anything above 3ms increases run time without significantly decreasing cpu usage time.

**Comment by khobbits** [comment]
Need me to rebase?

**Comment by laf** [comment]
Nope, only need to do that if it has a conflict.

**Comment by khobbits** [comment]
Okie, some people don't like large number of small commits ^^

**Comment by laf** [comment]
Ah you mean squash. It's fine, we'd probably ask you to if it was 50+ but it keeps history so no big deal.

**Comment by laf** [comment]
Looks good to me so :+1: 

Seems to spike the cpu but only at the start when I assume it's finding all rrd files to check.

Would you mind updating the http://docs.librenms.org/Support/Install%20Validation/ docs?

**Comment by khobbits** [comment]
Need anything else from me for this?

**Comment by f0o** [comment]
trigger ci

**Comment by f0o** [comment]
trigger ci

**Comment by f0o** [comment]
Doesnt seem to work on the CI:
```
root@2908:/opt/librenms# file rrd/test/broken.rrd 
rrd/test/broken.rrd: empty
```
```
root@2908:/opt/librenms# ./validate.php -r rrdcheck
[FAIL]    You need to run this script as root
fatal: Not a git repository (or any of the parent directories): .git
Version info:
[WARN]    Your install is out of date: 
DB Schema: 98
PHP: 5.6.17-0+deb8u1
MySQL: 10.0.22-MariaDB-0+deb8u1-log
RRDTool: 1.4.8
SNMP: NET-SNMP version: 5.7.2.1

[WARN]    You don't have $config["user"] set, this most likely needs to be set to librenms
[OK]      Database connection successful
[WARN]    Your RRD directory is owned by root, please consider changing over to user a non-root user
[WARN]    Your RRD directory is not set to 0775, please check our installation instructions
[OK]      fping has been found
```

**Pull request merged**